### PR TITLE
[ansible] Fix docker shell plugin

### LINKED ISSUE #1243: Docker shell module failed in the latest sonic-mgmt
**Description**
The task with sell type docker always fails on the latest sonic-mgmt, example:
```
- name: Gather information from lldp
  lldp:
  vars:
    ansible_shell_type: docker
    ansible_python_interpreter: docker exec -i lldp python
```

**Steps to reproduce the issue:**
1. run dip_sip or lag_2 CT 
2.
3.

**Describe the results you received:**
```
TASK [test : Gathering peer VM information from lldp] **************************
task path: /var/user/jenkins/bfn-sonic-mgmt/ansible/roles/test/tasks/lag_2.yml:26
Thursday 28 November 2019  10:28:47 +0000 (0:00:00.263)       0:00:26.753 ***** 
The full traceback is:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 145, in run
    res = self._execute()
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 612, in _execute
    self._set_connection_options(variables, templar)
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 1012, in _set_connection_options
    self._set_plugin_options('shell', final_vars, templar, task_keys)
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 978, in _set_plugin_options
    plugin.set_options(task_keys=task_keys, var_options=options)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/shell/__init__.py", line 70, in set_options
    env = self.get_option('environment')
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/__init__.py", line 60, in get_option
    raise KeyError(to_native(e))
KeyError: 'Requested entry (plugin_type: shell plugin: docker setting: environment ) was not defined in configuration.'

fatal: [cab18-2-dut]: FAILED! => {
    "msg": "Unexpected failure during module execution.", 
    "stdout": ""
}

```

**Describe the results you expected:**

**Additional information you deem important:**

    **Output of `show version`:**

    ```
SONiC Software Version: SONiC.HEAD.488-dirty-20191127.082217
Distribution: Debian 9.11
Kernel: 4.9.0-9-2-amd64
Build commit: 7622a30d
Build date: Wed Nov 27 11:15:51 UTC 2019
Built by: johnar@jenkins-worker-11

    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

### PR DESCRIPTION
**Pull request by msosyak**
### Description of PR

Update docker shell plugin to be compatible with ansible 2.8.7
Summary:
Fixes #1243

### Type of change

- [x] Bug fix
- [] Testbed and Framework(new/improvement)
- [] Test case(new/improvement)

#### How did you do it?
Add a description of the plugin. https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.4.html#configuration
Fixed base method call

#### How did you verify/test it?
Run few testcases
#### Any platform specific information?
N/A
#### Supported testbed topology if it's a new test case?
N/A

### CODE REVIEW

**Comment by msosyak** [comment]
@lguohan @qiluo-msft Please review. This is the fix for https://github.com/Azure/sonic-mgmt/issues/1243

File: ansible/shell_plugins/docker.py [hunk 1]
```diff
@@ -1,6 +1,17 @@
 from __future__ import (absolute_import, division)
 __metaclass__ = type
 
+DOCUMENTATION = '''
+name: sh
```
Comment (qiluo-msft) [inline ansible/shell_plugins/docker.py]: wrong name?

Comment (msosyak) [reply ansible/shell_plugins/docker.py]: Fixed

File: ansible/shell_plugins/docker.py [hunk 2]
```diff
@@ -51,13 +62,7 @@ def build_module_command(self, env_string, shebang, cmd, arg_path=None, rm_tmp=N
         pre = ''.join('docker exec {1} mkdir -p {0}; docker cp {0}/. {1}:{0}; '
```
Comment (qiluo-msft) [inline ansible/shell_plugins/docker.py]: >docker cp [](start = 53, length = 9)

If ansible command line has the environment variable `ANSIBLE_KEEP_REMOTE_FILES=0` (this is the default), we still need to remove the temporary file we `docker cp` into the container. I guess you need to override `ShellBase.remove` function.

Comment (msosyak) [reply ansible/shell_plugins/docker.py]: Done

File: ansible/shell_plugins/docker.py [hunk 3]
```diff
@@ -29,7 +41,22 @@ def join_path(self, *args):
 
         return super(ShellModule, self).join_path(*args)
 
-    def build_module_command(self, env_string, shebang, cmd, arg_path=None, rm_tmp=None):
+    def remove(self, path, recurse=False):
+        if not self.container:
+            self.container = self.get_option('ansible_python_interpreter').split()[3]
```
Comment (qiluo-msft) [inline ansible/shell_plugins/docker.py]: >self.get_option('ansible_python_interpreter').split()[3] [](start = 29, length = 56)

This logic is not robust.
If this case `not self.container` happens in your test? If not, we can just fallback to base class function.

Comment (msosyak) [reply ansible/shell_plugins/docker.py]: If I am right this module was made just for one thing, to execute tasks inside the container on the host. And it retrieves the container name from shebang in `build_module_command` which is not anything else as  `ansible_python_interpreter`. And for this plugin, it should look like `docker exec -i  <container> python`  what is not obvious. So I just put `get_option('ansible_python_interpreter')` under `if` to minimize `get_option` calls.

For any case changed this to the same approach as in `build_module_command`.

For me, it looks like is better to define mandatory `ansible_container` option for this plugin, change `remove` and `build_module_command` to use it as a container name source and replace 
```
vars:
    ansible_shell_type: docker
    ansible_python_interpreter: docker exec -i swss python
```
to 
```
vars:
    ansible_shell_type: docker
    ansible_container: swss
```
I could do this in separate PR if you accept.

**Pull request merged**